src/rulesDeploy.ts

```diff
@@ -100,6 +108,48 @@ export class RulesDeploy {
     return { latestName, latestContent };
   }
 
+  async checkStorageRulesIamPermissions(rulesContent: string | undefined): Promise<void> {
+    if (rulesContent?.match(CROSS_SERVICE_FUNCTIONS) === null) {
```

Comment:
Is it necessary to check for `null` explicitly? Would this be sufficent?

```suggestion
    if (!rulesContent?.match(CROSS_SERVICE_FUNCTIONS)) {
```

Reply:
I'd like to keep this one as-is. This matches the logic we have in the Console. Based on my understanding of `match()`, it _shouldn't_ matter, but since there is a subtle semantic difference between strict null check and a general falsey check, I want to keep the two in sync